EIPS/eip-xx.md

```diff
@@ -0,0 +1,117 @@
+
+What if all those Java developers (currently in the millions), could be onboarded into the Ethereum Ecosystem?
+
+![](https://github.com/jeyakatsa/EIPs/blob/master/assets/eip-xx/Java-Abstraction-Visualization-display.jpg)
+
+## Specification
+
+##### Smart-Contract Storage example in *Solidity*:
+```solidity
+pragma solidity >=0.4.16 <0.9.0;
+
+contract SimpleStorage {
+    uint storedData;
+
+    function set(uint x) public {
+        storedData = x;
+    }
+
+    function get() public view returns (uint) {
+        return storedData;
+    }
+}
+```
+
+##### Smart-Contract Storage example in *Java*:
+```java
+public class SimpleStorage {
+    private Uint256 storedData;
+
+    public void setStoredData (Uint256 storedData) {
+        this.storedData = storedData;
+    }
+
+    public Uint256 getStoredData () {
+        return storedData;
+    }
+}
+```
+
+## Rationale
+
+### Solidity Keywords to Java Dependency conversion process:
+The `uint` keyword in Solidity essentially represents a pre-packaged library containing a 256 bit byte. In Java, a dependency is created to suplement for the Solidity keyword as follows:
+
+##### `Uint256` Java Dependency (in place of `uint` Solidity Keyword) example:
+```java
+public interface Uint256 {
+
+    static byte[] ivBytes = new byte[256];
+    static int iterations = 65536;
+    static int keySize = 256;
+    static byte[] uint = new byte[256];
+
+    public default void Uint256() throws Exception {
+        decrypt();
+    }
+
+    public static void decrypt() throws Exception {
+
+        char[] placeholderText = new char[0];
+
+        SecretKeyFactory skf = SecretKeyFactory.getInstance("PBKDF2WithHmacSHA1");
+        PBEKeySpec spec = new PBEKeySpec(placeholderText, Uint256.uint, iterations, keySize);
+        SecretKey secretkey = skf.generateSecret(spec);
+        SecretKeySpec secretSpec = new SecretKeySpec(secretkey.getEncoded(), "AES");
+
+        Cipher cipher = Cipher.getInstance("AES/CBC/PKCS5Padding");
+        cipher.init(Cipher.DECRYPT_MODE, secretSpec, new IvParameterSpec(ivBytes));
+
+        byte[] decryptedTextBytes = null;
+
+        try {
+            decryptedTextBytes = cipher.doFinal();
+        }   catch (IllegalBlockSizeException e) {
+            e.printStackTrace();
+        }   catch (BadPaddingException e) {
+            e.printStackTrace();
+        }
+
+        decryptedTextBytes.toString();
+    }
+}
+```
+
+## Backwards Compatibility
+These Smart-Contract Java abstractions are meant to be compatible with the overall Ethereum Execution and Consensus layers respectfully.
```

Comment:
I'm not sure how this would be compatible with the consensus layer given it has no execution?

Reply:
Still working on the execution, will return once that's implemented. Thank you @lightclient